### ISSUE TITLE
Support to new Header API

### ISSUE THREAD

**Issue by carlos-verdes** [etorreborre/specs2-http4s#4]
I took the matchers from this repo and adapted to the new Header API.

If this helps to add that support I share one snippet:
```scala
  def containHeader[H](hs: H)(implicit H: Header.Select[H]): Matcher[Message[F]] =
    beSome(hs) ^^ { (m: Message[F]) =>
      m.headers.get[H](H).asInstanceOf[Option[H]].aka("the particular header")
    }
```

**Comment by etorreborre** [comment]
Thanks, I just released `1.1.0` with an upgrade to `https4s-1.0.0-M23`

**Comment by carlos-verdes** [comment]
Amazing, thanks!
